### DISCUSSION TITLE
ROCm in Raspberry PI5

### DISCUSSION THREAD

**Discussion by tanreinama** [ROCm/ROCm#2988] [Ideas]
In my environment, RadeonRX480 is running on RaspberryPI5.
https://github.com/geerlingguy/raspberry-pi-pcie-devices
I used this repository as a reference.

And I found that the ROCm source code is compatible with AARCH64.
https://github.com/ROCm/ROCm/issues/231

Therefore, I think I can run ROCm on RaspberryPI and use it for edge AI, but it doesn't work.

I downloaded the source code using repo, compiled and installed rocm-core, but I'm having trouble installing the kernel module.

I feel like I'm simply not doing it enough.
Any ideas?

**Comment by fluidnumericsJoe** [comment]
What does `uname -r` show for your Linux kernel version?

Which raspberry pi has a discrete AMD GPU? Integrated AMD graphics won't work with ROCm

Edit: checked the link. I didn't realize Raspberry Pi now has PCIe! Quite neat. Disregard my last question about the discrete GPU

What are the specific errors you're seeing when building the AMDGPU drivers?

**Comment by tanreinama** [comment]
```
$ uname -r
6.1.61-v8_16k-amd+
```
The "-amd+" part is the SUSFIX I added when I compiled the kernel.
Not integrated graphics.
I installed an external RadeonRX480 on the PCIe port of RaspberryPI5 via a PCIe->M.2 adapter and M.2->PCIe x16 riser card.
The power supply is also wired and the graphics board is working.

Simply, I don't know how to compile and install ROCm from source code.
Once please forgetting about RaspberryPI, could you tell me how to install ROCm from the source code (not apt-get) on PC?
When you sync using repo, many directories are created. I was able to build some of them using cmake, but I don't know how to build the whole thing.

extra hw imformation.
```
$ lspci
0000:00:00.0 PCI bridge: Broadcom Inc. and subsidiaries Device 2712 (rev 21)
0000:01:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Ellesmere [Radeon RX 470/480/570/570X/580/580X/590] (rev c7)
0000:01:00.1 Audio device: Advanced Micro Devices, Inc. [AMD/ATI] Ellesmere HDMI Audio [Radeon RX 470/480 / 570/580/590]
0001:00:00.0 PCI bridge: Broadcom Inc. and subsidiaries Device 2712 (rev 21)
0001:01:00.0 Ethernet controller: Device 1de4:0001
```

```
 $ dmesg|grep amd
[    0.000000] Linux version 6.1.61-v8_16k-amd+ (root@raspberrypi) (gcc (Debian 12.2.0-14) 12.2.0, GNU ld (GNU Binutils for Debian) 2.40) #1 SMP PREEMPT Sun Mar 17 14:29:09 JST 2024
[    0.729754] usb usb1: Manufacturer: Linux 6.1.61-v8_16k-amd+ xhci-hcd
[    0.730135] usb usb2: Manufacturer: Linux 6.1.61-v8_16k-amd+ xhci-hcd
[    0.789729] usb usb3: Manufacturer: Linux 6.1.61-v8_16k-amd+ xhci-hcd
[    0.790067] usb usb4: Manufacturer: Linux 6.1.61-v8_16k-amd+ xhci-hcd
[   15.025562] [drm] amdgpu kernel modesetting enabled.
[   15.025664] amdgpu: CRAT table disabled by module option
[   15.025667] amdgpu: DSDT table not found for OEM information
[   15.025669] amdgpu: IO link not available for non x86 platforms
[   15.025671] amdgpu: Virtual CRAT table created for CPU
[   15.025686] amdgpu: Topology: Add CPU node
[   15.025860] amdgpu 0000:01:00.0: enabling device (0000 -> 0002)
[   15.237875] amdgpu 0000:01:00.0: amdgpu: Fetched VBIOS from ROM BAR
[   15.237883] amdgpu: ATOM BIOS: 113-V34111-F1
[   15.237923] amdgpu 0000:01:00.0: amdgpu: Trusted Memory Zone (TMZ) feature not supported
[   15.237927] amdgpu 0000:01:00.0: amdgpu: PCIE atomic ops is not supported
[   15.373214] amdgpu 0000:01:00.0: BAR 2: releasing [mem 0x1810000000-0x18101fffff 64bit pref]
[   15.373220] amdgpu 0000:01:00.0: BAR 0: releasing [mem 0x1800000000-0x180fffffff 64bit pref]
[   15.373255] amdgpu 0000:01:00.0: BAR 0: assigned [mem 0x1800000000-0x19ffffffff 64bit pref]
[   15.373265] amdgpu 0000:01:00.0: BAR 2: assigned [mem 0x1a00000000-0x1a001fffff 64bit pref]
[   15.373288] amdgpu 0000:01:00.0: amdgpu: VRAM: 8192M 0x000000F400000000 - 0x000000F5FFFFFFFF (8192M used)
[   15.373291] amdgpu 0000:01:00.0: amdgpu: GART: 256M 0x000000FF00000000 - 0x000000FF0FFFFFFF
[   15.373342] [drm] amdgpu: 8192M of VRAM memory ready
[   15.373345] [drm] amdgpu: 4026M of GTT memory ready.
[   15.393599] amdgpu: hwmgr_sw_init smu backed is polaris10_smu
[   15.871888] kfd kfd: amdgpu: skipped device 1002:67df, PCI rejects atomics 730<0
[   15.871907] amdgpu 0000:01:00.0: amdgpu: SE 4, SH per SE 1, CU per SH 9, active_cu_number 36
[   15.875210] amdgpu 0000:01:00.0: amdgpu: Using BACO for runtime pm
[   15.880076] [drm] Initialized amdgpu 3.49.0 20150101 for 0000:01:00.0 on minor 2
[   15.885274] amdgpu 0000:01:00.0: [drm] Cannot find any crtc or sizes
```

**Comment by fortunerains** [comment]
How's it going now

**Comment by zichguan-amd** [comment]
Hi @tanreinama, building ROCm from source is time & resource intensive, you can refer to the docs [here](https://github.com/ROCm/ROCm?tab=readme-ov-file#build-rocm-from-source). We recommend using the package manager if you are on a supported OS as per the [quick start guide](https://rocm.docs.amd.com/projects/install-on-linux/en/develop/install/quick-start.html). 
RX480 is not supported by the latest ROCm release anymore, you'd have better chance using one of the supported GPUs [here](https://rocm.docs.amd.com/projects/install-on-linux/en/develop/reference/system-requirements.html). You may also try an older version of ROCm. We won't be able to help with Raspberry Pi since it's not supported.

**Reply by tanreinama** [reply]
Thanks for the info. Unfortunately, the RX480 was the latest one that worked with the RaspberryPi. In conclusion, it seems like it will be difficult to use the RocM with the RaspberryPi.
